examples/yacat/yacat.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+import asyncio
+from datetime import timedelta
+import pathlib
+import sys
+
+from yapapi.log import enable_default_logger, log_summary, log_event_repr  # noqa
+from yapapi.runner import Engine, Task, vm
+from yapapi.runner.ctx import WorkContext
+
+
+# For importing `utils.py`:
+script_dir = pathlib.Path(__file__).resolve().parent
+parent_directory = script_dir.parent
+sys.stderr.write(f"Adding {parent_directory} to sys.path.\n")
+sys.path.append(str(parent_directory))
+import utils  # noqa
+
+
+def write_hash(hash):
+    with open("in.hash", "w") as f:
+        f.write(hash)
+
+
+def write_keyspace_check_script(mask):
+    with open("keyspace.sh", "w") as f:
+        f.write(f"hashcat --keyspace -a 3 {mask} -m 400 > /golem/work/keyspace.txt")
+
+
+def read_keyspace():
+    with open("keyspace.txt", "r") as f:
+        return int(f.readline())
+
+
+def read_password(ranges):
+    for r in ranges:
+        f = open(f"hashcat_{r}.potfile", "r")
+        line = f.readline()
+        split_list = line.split(":")
+        if len(split_list) >= 2:
+            return split_list[1]
+    return None
+
+
+async def main(args):
+    package = await vm.repo(
+        image_hash="2c17589f1651baff9b82aa431850e296455777be265c2c5446c902e9",
+        min_mem_gib=0.5,
+        min_storage_gib=2.0,
+    )
+
+    async def worker_check_keyspace(ctx: WorkContext, tasks):
```

Comment:
Type annotations are used only here and when you declare `range`. I'd use them everywhere or just remove them.

Reply:
leaving it as it is as discussed